src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/CrossCaseTypeSearchRequest.java

```diff
@@ -84,8 +87,15 @@ public boolean hasAliasField(SearchAliasField searchAliasField) {
         return aliasFields.stream().anyMatch(aliasField -> aliasField.equalsIgnoreCase(searchAliasField.getId()));
     }
 
+    public boolean isConsolidationQuery() {
+        return consolidationQuery;
+    }
+
     private void addMetadataSourceFields() {
         if (elasticsearchRequest.hasSource()) {
+            if (elasticsearchRequest.getSource() instanceof BooleanNode) {
+                return;
+            }
```

Comment:
not sure I understand this. When _source=true we in theory we have to addMetadataSourceFields. But the current code is not?

Reply:
I think this logic (once corrected) could be better placed inside the `elasticsearchRequest.hasSource()` method. And rename that method to:
`elasticsearchRequest.hasSourceFields()`

Reply:
The original code is doing custom manipulation on source assuming that source is always an Array[].  **If the source = false (valid elastic query)** will generate a Runtime Exception.    Bub in dev.    The if statement allows the current implementation to accept source = false and let elasticsearch to processes it . Nicer implementation can be done. we will discuss later.

Reply:
DONE